### PR TITLE
[GH-5674] Upgrade H2O to 3.44.0.1

### LINKED ISSUE #5674: Upgrade H2O to 3.44.0.1

### PR DESCRIPTION
**Pull request by krasinski**
closes #5674

### CODE REVIEW

**Review by tomasfryda — APPROVED** [review]
LGTM! Thank you @krasinski !

File: py-scoring/README.rst [hunk 1]
```diff
@@ -7,6 +7,7 @@ This package contains just functionality for scoring with Sparkling Water, H2O-3
 
 Documentation describing scoring with H2O-3 MOJO models is located at:
 
+- For Spark 3.5 - https://docs.h2o.ai/sparkling-water/3.5/latest-stable/doc/deployment/load_mojo.html
```
Comment (mn-mikke) [inline py-scoring/README.rst:10]: Shouldn't the changes in this file be rather a part of https://github.com/h2oai/sparkling-water/pull/5664?

File: py/README.rst [hunk 2]
```diff
@@ -8,6 +8,7 @@ to use this package for scoring with Driverless AI MOJO models.
 
 PySparkling Documentation is hosted at our documentation page:
 
+- For Spark 3.5 - http://docs.h2o.ai/sparkling-water/3.5/latest-stable/doc/pysparkling.html
```
Comment (mn-mikke) [inline py/README.rst:11]: Shouldn't the changes in this file be rather a part of https://github.com/h2oai/sparkling-water/pull/5664?

**Review by mn-mikke — APPROVED** [review]
Extra changes, otherwise LGTM.

**Pull request merged**